internal/common/normalattackstate.go

```diff
@@ -134,3 +140,146 @@ func Get5PercentN0Delay(activeChar *character.CharWrapper) int {
 
 	return percentDelay5[activeCharKey]
 }
+
+func Get0PercentN0Delay(activeChar *character.CharWrapper) int {
+	// TODO: Collect data for this
+	return 0
+}
+
+type NAHook struct {
+	c           *character.CharWrapper
+	abilName    string
+	Core        *core.Core
+	abilKey     string
+	abilProcICD int
+	abilICDKey  string
+	abilTickSrc int
+	abilHookSrc int
+	delayFunc   func(*character.CharWrapper) int
+	summonFunc  func()
+}
+
+func NewNAHook(c *character.CharWrapper, Core *core.Core, abilName string, abilKey string, abilProcICD int, abilICDKey string, delayFunc func(*character.CharWrapper) int, summonFunc func()) *NAHook {
+	return &NAHook{
+		c:           c,
+		abilName:    abilName,
+		Core:        Core,
+		abilKey:     abilKey,
+		abilProcICD: abilProcICD,
+		abilICDKey:  abilICDKey,
+		abilTickSrc: 0,
+		abilHookSrc: 0,
+		delayFunc:   delayFunc,
+		summonFunc:  summonFunc,
+	}
+
+}
+
+func (w *NAHook) NAStateHook() {
+	w.Core.Events.Subscribe(event.OnAttack, func(args ...interface{}) bool {
+		//check if buff is up
+		if !w.c.StatusIsActive(w.abilKey) {
+			return false
+		}
+		w.abilHookSrc = w.Core.F
+		delay := w.delayFunc(w.Core.Player.ActiveChar())
+		w.Core.Log.NewEvent(fmt.Sprintf("%v delay on state change", w.abilName), glog.LogCharacterEvent, w.c.Index).
+			Write("delay", delay)
+		// This accounts for the delay in n0 timing needed to trigger
+
+		if delay > 0 {
+
+			w.Core.Tasks.Add(w.naStateDelayFuncGen(w.Core.F), delay)
+		} else {
```

Comment:
return inside this `if` and remove `else` to reduce code depth

NIT: remove blank line